js/blocks/controls/number.js

```diff
@@ -28,7 +28,7 @@ const BlockLabNumberControl = ( props, field, block ) => {
 			defaultValue={field.default}
 			value={attr[ field.name ]}
 			onChange={numberControl => {
-				attr[ field.name ] = numberControl
+				attr[ field.name ] = Number( numberControl )
```

Comment:
Incorrect values were being saved because they were being passed to the validation function as a string.